file/prefetch_test.cc

```diff
@@ -3496,18 +3578,28 @@ TEST_P(FSBufferPrefetchTest, FSBufferPrefetchStatsInternals) {
     ASSERT_EQ(std::get<1>(buffer_info[0]), 12288);
   }
 
-  ASSERT_TRUE(
-      fpb.TryReadFromCache(IOOptions(), r.get(), 12288, 4096, &result, &s));
+  ASSERT_TRUE(fpb.TryReadFromCache(IOOptions(), r.get(), 12288, 4096, &result,
+                                   &s, for_compaction));
   ASSERT_EQ(s, Status::OK());
-  ASSERT_EQ(stats->getAndResetTickerCount(PREFETCH_HITS), 1);
-  ASSERT_EQ(stats->getAndResetTickerCount(PREFETCH_BYTES_USEFUL),
-            4096);  // 12288-16384
+  if (!for_compaction) {
```

Comment:
why !for_compaction?

Reply:
I can add a comment in the code for this. 

We do not update the prefetch stats for compaction reads https://github.com/facebook/rocksdb/blob/main/file/file_prefetch_buffer.cc#L856-L857, so I need this check to keep the tests passing for all the different test parameter variations
```cpp
  } else if (!for_compaction) {
    UpdateStats(/*found_in_buffer=*/true, n);
  }
```

I don't know why exactly we don't do this. @anand1976 may know. I would guess that we don't want to "contaminate" the prefetch stats for non-compaction reads (`PREFETCH_HITS` and `PREFETCH_BYTES_USEFUL`) which we presumably are more interested in. I think we could have defined a separate set of prefetch stats for compaction reads specifically, but maybe we just did not care because the compaction reads are all background ops anyways

Reply:
Update: I changed the logic to just update stats for both compaction and non-compaction reads. I think it will make our lives easier